Mathlib/Analysis/SpecialFunctions/Pow/Real.lean

```diff
@@ -746,6 +752,13 @@ theorem rpow_left_injOn {x : ℝ} (hx : x ≠ 0) : InjOn (fun y : ℝ => y ^ x)
 lemma rpow_left_inj (hx : 0 ≤ x) (hy : 0 ≤ y) (hz : z ≠ 0) : x ^ z = y ^ z ↔ x = y :=
   (rpow_left_injOn hz).eq_iff hx hy
 
+lemma rpow_right_inj (hx₀ : 0 < x) (hx₁ : x ≠ 1) : x ^ y = x ^ z ↔ y = z := by
+  refine ⟨fun H ↦ ?_, fun H ↦ by rw [H]⟩
+  rcases hx₁.lt_or_lt with h | h
+  · exact le_antisymm ((rpow_le_rpow_left_iff_of_base_lt_one hx₀ h).mp H.symm.le) <|
+      (rpow_le_rpow_left_iff_of_base_lt_one hx₀ h).mp H.le
+  · exact le_antisymm ((rpow_le_rpow_left_iff h).mp H.le) ((rpow_le_rpow_left_iff h).mp H.symm.le)
```

Comment:
If you move this down ~100 lines, then you can golf it using `strictMono_rpow_of_base_gt_one` and `strictAnti_rpow_of_base_lt_one`.

Reply:
OK, thanks! (It did seem to make sense to put it after the analogous `left_inj` statement...)